lib/dependabot/shared_helpers.rb

```diff
@@ -62,35 +63,60 @@ def self.in_a_forked_process
     end
 
     class HelperSubprocessFailed < StandardError
-      def initialize(message, command)
+      def initialize(message, error_context:)
         super(message)
-        @command = command
+        @error_context = error_context
+        @command = error_context[:command]
       end
 
       def raven_context
-        { fingerprint: [@command] }
+        { fingerprint: [@command], extra: @error_context }
       end
     end
 
+    # rubocop:disable Metrics/MethodLength
     def self.run_helper_subprocess(command:, function:, args:, env: nil,
-                                   popen_opts: {})
-      raw_response = nil
-      popen_args = [env, command, "w+"].compact
-      IO.popen(*popen_args, popen_opts) do |process|
-        process.write(JSON.dump(function: function, args: args))
-        process.close_write
-        raw_response = process.read
+                                   std_err_to_std_out: false)
+      json_response = nil
+      stderr_output = nil
+      process_exit_value = nil
+
+      start = Time.now
+      Open3.popen3(*[env, command].compact) do |stdin, stdout, stderr, wait_thr|
```

Comment:
Just saw this disclaimer in the Ruby docs about `open3`:

> You should be careful to avoid deadlocks. Since pipes are fixed length buffers, Open3.popen3(“prog”) {|i, o, e, t| o.read } deadlocks if the program generates too much output on stderr. You should read stdout and stderr simultaneously (using threads or IO.select). However, if you don't need stderr output, you can use Open3.popen2. If merged stdout and stderr output is not a problem, you can use Open3.popen2e. If you really need stdout and stderr output as separate strings, you can consider Open3.capture3.

Have you looked into `capture3`?